### PR TITLE
Add API testcase for virt-who-configure plugin testing

### PR DESCRIPTION
**Pull request by hkx303**
1. Add one API testcase for virt-who-configure plugin testing
2. Update deploy_validation() in utils.py so that there will have enough time for virt-who restart process

### CODE REVIEW

**Comment by hkx303** [comment]
Test Result:
(venv) [hkx303@kuhuang virtwho]$ pytest test_api_virtwho.py 
2019-08-26 09:16:13 - conftest - DEBUG - Registering custom pytest_configure

==================================== test session starts =====================================
platform linux -- Python 3.7.3, pytest-4.0.2, py-1.8.0, pluggy-0.12.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/hkx303/Documents/CI/robottelo/robottelo, inifile:
plugins: services-1.3.1, forked-1.0.2, mock-1.10.4, cov-2.7.1, xdist-1.29.0
collecting ... 2019-08-26 09:16:13 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                             

test_api_virtwho.py .

=========================== 1 passed, 3 warnings in 103.51 seconds ===========================

File: tests/foreman/virtwho/test_api_virtwho.py [hunk 1]
```diff
@@ -14,31 +14,97 @@
...
-        4. Virtual sku can be generated and attached
-        5. Config can be deleted
-
-    :CaseAutomation: notautomated
-    """
+from nailgun import entities
+from fauxfactory import gen_string
+from robottelo.test import APITestCase
+from robottelo.config import settings
+from robottelo.decorators import tier1
```
Comment (vijay8451) [inline tests/foreman/virtwho/test_api_virtwho.py]: Please set in order to alphabetically .. ^^

File: tests/foreman/virtwho/test_api_virtwho.py [hunk 2]
```diff
@@ -14,31 +14,97 @@
...
+
+class VirtWhoConfigApiTestCase(APITestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(VirtWhoConfigApiTestCase, cls).setUpClass()
+        org = entities.Organization().search(
+            query={'search': 'name="Default Organization"'}
+        )[0]
+        cls.org = org.read()
```
Comment (vijay8451) [inline tests/foreman/virtwho/test_api_virtwho.py]: Why do you require to `read` the org again .. can you please use as below :
```
 cls.org = entities.Organization().search(
            query={'search': 'name="Default Organization"'}
    :expectedresults:
        )[0]
```

File: tests/foreman/virtwho/test_api_virtwho.py [hunk 3]
```diff
@@ -14,31 +14,97 @@
...
+from nailgun import entities
+from fauxfactory import gen_string
+from robottelo.test import APITestCase
+from robottelo.config import settings
+from robottelo.decorators import tier1
+
+from .utils import (
+    deploy_configure_by_command,
+    get_configure_command,
+)
```
Comment (vijay8451) [inline tests/foreman/virtwho/test_api_virtwho.py]: Can we not write the utils in API as well .. as that would be far faster then CLI ?

File: tests/foreman/virtwho/test_api_virtwho.py (hunk 1 above)
Comment (hkx303) [reply tests/foreman/virtwho/test_api_virtwho.py]: ok, thanks

File: tests/foreman/virtwho/test_api_virtwho.py (hunk 2 above)
Comment (hkx303) [reply tests/foreman/virtwho/test_api_virtwho.py]: yes, thanks

File: tests/foreman/virtwho/test_api_virtwho.py [hunk 4]
```diff
@@ -14,31 +14,97 @@
...
+            host = entities.Host().search(
+                query={'search': "{0}".format(hostname)})[0]
+            product_subscription = entities.Subscription().search(
+                query={'search': '{0}'.format(sku)})[0]
+            entities.HostSubscription(host=host.id).add_subscriptions(
+                data={'subscriptions': [{
+                              'id': product_subscription.id,
+                              'quantity': 1}]})
+            result = entities.Host().search(
+                query={'search': '{0}'.format(hypervisor_name)})[0].read_json()
```
Comment (eko999) [inline tests/foreman/virtwho/test_api_virtwho.py]: It should be searching `hostname`, we not only need to check the hypervisor's subscription status, the guest need to check as well.

File: tests/foreman/virtwho/test_api_virtwho.py (hunk 3 above)
Comment (hkx303) [reply tests/foreman/virtwho/test_api_virtwho.py]: We only have 2 methods to deploy the virt-who-config plugin, via hammer command or execute the configuration script. So though we testing virt-who-config by API now, we still should use CLI to deploy it.

File: tests/foreman/virtwho/test_api_virtwho.py (hunk 4 above)
Comment (hkx303) [reply tests/foreman/virtwho/test_api_virtwho.py]: My bad, that's what I mean, thanks

**Pull request merged**